JsonSubTypes/JsonSubtypes.cs

```diff
@@ -444,9 +453,30 @@ private static object ThreadStaticReadObject(JsonReader reader, JsonSerializer s
             }
         }
 
+        private static IEnumerable<object> GetAttributes(TypeInfo typeInfo)
+        {
+#if NET35
+            if (_attributesCache.TryGetValue(typeInfo, out var res))
```

Comment:
You can't do this outside of the lock, the dictionary is not thread safe and might be adding an item at the same time

Reply:
Fixed, I hope that my fix didn't add additional issues